system_test/aest_sync_SUITE.erl

```diff
@@ -334,6 +337,103 @@ stop_and_continue_sync(Cfg) ->
             ?assertEqual(C1, B2)
     end.
 
+tx_pool_sync(Cfg) ->
+    NodeStartupTime = proplists:get_value(node_startup_time, Cfg),
+
+    setup_nodes([#{ name    => node1,
+                    peers   => [node2],
+                    backend => aest_docker,
+                    source  => {pull, "aeternity/epoch:local"}
+                  },
+                 #{ name    => node2,
+                    peers   => [node1],
+                    backend => aest_docker,
+                    source  => {pull, "aeternity/epoch:local"}
+                  }], Cfg),
+
+    start_node(node1, Cfg),
+    wait_for_value({height, 0}, [node1], NodeStartupTime, Cfg),
+
+    %% Let's post a bunch of transactions, preferrably some valid
+    %% and some "not yet valid"
+
+    Patron = #{ pubkey => <<206,167,173,228,112,201,249,157,157,78,64,8,128,168,111,29,73,187,68,75,98,241,26,158,187,100,187,207,235,115,254,243>>,
+                privkey => <<230,169,29,99,60,119,207,87,113,50,157,51,84,179,188,239,27,197,224,50,196,61,112,182,211,90,249,35,206,30,183,77,206,167,173,228,112,201,249,157,157,78,64,8,128,168,111,29,73,187,68,75,98,241,26,158,187,100,187,207,235,115,254,243>>
+              },
+
+    %% Add 5 valid spend transactions
+    ValidTxs = add_spend_txs(node1, Patron, 5, 1),
+    %% Add 10 invalid (nonce_too_high) spend transactions
+    InvalidTxs = add_spend_txs(node1, Patron, 5, 7),
+    add_spend_txs(node1, Patron, 5, 15),
+
+
+    %% Check that the valid transactions made it to the chain.
+    #{ reciever := RecvAccount, amount := Amount } = lists:last(ValidTxs),
+    wait_for_value({balance, aec_base58c:encode(account_pubkey, RecvAccount), Amount},
```

Comment:
The receiver account is distinct in each spend, so this is checking that last tx included. And the last tx has highest nonce, hence its inclusion implies inclusion of txs with lower nonce too. Ok.